compiler/rustc_ast/src/attr/mod.rs

```diff
@@ -425,7 +425,7 @@ impl MetaItem {
         let kind = MetaItemKind::from_tokens(tokens)?;
         let hi = match kind {
             MetaItemKind::NameValue(ref lit) => lit.span.hi(),
-            MetaItemKind::List(..) => list_closing_paren_pos.unwrap_or(path.span.hi()),
+            MetaItemKind::List(..) => list_closing_paren_pos.unwrap_or_else(|| path.span.hi()),
```

Comment:
```suggestion
            MetaItemKind::List(..) => list_closing_paren_pos.unwrap_or(path.span.hi()),
```
Removing this and other "noisy" cases, as to improve readability.